metricflow/sql/sql_exprs.py

```diff
@@ -1314,6 +1320,65 @@ def matches(self, other: SqlExpressionNode) -> bool:  # noqa: D102
         return self.count == other.count and self.granularity == other.granularity and self._parents_match(other)
 
 
+@dataclass(frozen=True, eq=False)
+class SqlAddTimeExpression(SqlExpressionNode):
```

Comment:
Would be good if we added some tests for this with the different engine rendering

Reply:
I'll have some up the stack! But it felt very annoying to add those tests without having the logic to render them in a full SQL query yet.

Reply:
Ahh i see. Though, you should be able to test this node in isolation here https://github.com/dbt-labs/metricflow/blob/main/tests_metricflow/sql/test_engine_specific_rendering.py, but if you already have these tests up the stack on this, totally fine too!

Reply:
Good idea I'll add those.